apis/v1alpha1/httproute_types.go

```diff
@@ -319,20 +306,6 @@ type HTTPRouteMatch struct {
 	//
 	// +optional
 	Headers *HTTPHeaderMatch `json:"headers"`
-
-	// ExtensionRef is an optional, implementation-specific extension to the
-	// "match" behavior.  The resource may be "configmap" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myroutematchers" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
-	//
-	// Support: custom
-	//
-	// +optional
-	ExtensionRef *LocalObjectReference `json:"extensionRef,omitempty"`
```

Comment:
The reasoning for removing this extension point stems from the case that if we chose to keep this extension point, then the routing definition will be defined in HTTPRoute but the match will be defined in another CRD. This can get cumbersome for users to read and maintain.

On another note, if we remove this extension point, then does having extension points inside TCPRoute, UDPRoute and TLSRoute at the same level make sense?

Reply:
I'm not quite as sure about this removal since we have extension points at the same place on other routes and this would be harder to replicate through other methods like name-mapping or annotations. With that said, one could always create/support custom routes for different forms of matching.

Reply:
I know this is a little bit of a contention point and I'm not sure about this either.
I've removed this from the scope of this PR right now.
Tracking this in https://github.com/kubernetes-sigs/service-apis/issues/359.